From 2176a45a5ee6adb58853770254e3d2ab24229962 Mon Sep 17 00:00:00 2001
From: Yuan Can <yuancan@huawei.com>
Date: Fri, 15 Apr 2022 07:34:11 +0000
Subject: [PATCH 2664/4585] [Huawei] mm/sharepool: Avoid NULL pointer
 dereference in mg_sp_group_add_task

Offering: HULK
ascend inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5DS9S

------------------------------------------------------

create_spg_node() may fail with NULL pointer returened, and in
the out_drop_spg_node path, the NULL pointer will be dereferenced
in free_spg_node().

Signed-off-by: Yuan Can <yuancan@huawei.com>
---
 mm/share_pool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index 8f267c649230..e2282e6ab239 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -1523,7 +1523,7 @@ int mg_sp_group_add_task(int pid, unsigned long prot, int spg_id)
 	if (unlikely(IS_ERR(node))) {
 		up_write(&spg->rw_lock);
 		ret = PTR_ERR(node);
-		goto out_drop_spg_node;
+		goto out_drop_group;
 	}
 
 	ret = insert_spg_node(spg, node);
-- 
2.17.1

